test/AxelarGateway.js

```diff
@@ -269,21 +349,74 @@ describe('AxelarGateway', () => {
             const newGatewayImplementationCodeHash = await getBytecodeHash(newGatewayImplementation);
             const params = '0x';
 
+            await expect(
+                gateway.connect(notGovernance).upgrade(newGatewayImplementation.address, newGatewayImplementationCodeHash, params),
+            ).to.be.revertedWithCustomError(gateway, 'NotGovernance');
+
             await expect(
                 gateway
-                    .connect(notGovernance)
+                    .connect(governance)
                     .upgrade(newGatewayImplementation.address, newGatewayImplementationCodeHash, params, getGasOptions()),
+            )
+                .to.emit(gateway, 'Upgraded')
+                .withArgs(newGatewayImplementation.address)
+                .to.not.emit(gateway, 'GovernanceTransferred')
+                .to.not.emit(gateway, 'OperatorshipTransferred');
+        });
+
+        it('should allow governance to upgrade to the correct implementation with new governance and mint limiter', async () => {
+            const newGatewayImplementation = await gatewayFactory.deploy(auth.address, tokenDeployer.address).then((d) => d.deployed());
+            const newGatewayImplementationCodeHash = await getBytecodeHash(newGatewayImplementation);
+
+            let params = '0x';
+
+            await expect(
+                gateway.connect(notGovernance).upgrade(newGatewayImplementation.address, newGatewayImplementationCodeHash, params),
             ).to.be.revertedWithCustomError(gateway, 'NotGovernance');
 
+            params = getWeightedProxyDeployParams(notGovernance.address, notGovernance.address, []);
+
+            await expect(
+                gateway
+                    .connect(governance)
+                    .upgrade(newGatewayImplementation.address, newGatewayImplementationCodeHash, params, getGasOptions()),
+            )
+                .to.emit(gateway, 'Upgraded')
+                .withArgs(newGatewayImplementation.address)
+                .to.emit(gateway, 'GovernanceTransferred')
+                .withArgs(governance.address, notGovernance.address)
+                .to.emit(gateway, 'MintLimiterTransferred')
+                .withArgs(governance.address, notGovernance.address);
+        });
+
+        it('should allow governance to upgrade to the correct implementation with new operators', async () => {
+            const newGatewayImplementation = await gatewayFactory.deploy(auth.address, tokenDeployer.address).then((d) => d.deployed());
+            const newGatewayImplementationCodeHash = await getBytecodeHash(newGatewayImplementation);
```

Comment:
```suggestion
            const newGatewayImplementationCodeHash = await getBytecodeHash(newGatewayImplementation, network.config.id);
```
can you add this everywhere this method is used?